tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -2923,6 +2923,52 @@ public function test_create_custom_dimension() {
 		$this->assertEquals( "/v1beta/properties/$property_id/customDimensions", $request_url['path'] );
 	}
 
+	public function test_create_custom_dimension_without_optional_params() {
+		$property_id = '123456789';
+
+		$this->fake_handler_and_invoke_register_method( $property_id );
+
+		$this->grant_scope( Analytics_4::EDIT_SCOPE );
+
+		$custom_dimension = array(
+			'displayName'   => 'Test Custom Dimension',
+			'parameterName' => 'googlesitekit_post_author',
+			'scope'         => 'EVENT',
+		);
+
+		$response = $this->analytics->set_data(
+			'create-custom-dimension',
+			array(
+				'propertyID'      => $property_id,
+				'customDimension' => $custom_dimension,
+			)
+		);
+
+		$this->assertNotWPError( $response );
+
+		$response_array = (array) $response;
+
+		// Assert that the keys exist.
+		$keys = array_keys( $custom_dimension );
+
+		foreach ( $keys as $key ) {
+			$this->assertArrayHasKey( $key, $response_array );
+		}
+
+		// Validate the response against the expected mock value.
+		foreach ( $custom_dimension as $key => $value ) {
+			$this->assertEquals( $value, $response_array[ $key ] );
+		}
```

Comment:
Asserting the response like this doesn't quite make sense, because it's hardwired to be an object which _does_ have the optional fields set:

https://github.com/google/site-kit-wp/blob/66c60822f669d225af698f1fb4aad769695a2e63/tests/phpunit/integration/Modules/Analytics_4Test.php#L3139-L3151

I'd suggest reorganising the test as follows:

```suggestion
		// Create a custom dimension with only the required fields present.
		$raw_custom_dimension = array(
			'displayName'   => 'Test Custom Dimension',
			'parameterName' => 'googlesitekit_post_author',
			'scope'         => 'EVENT',
		);

		$this->fake_handler_and_invoke_register_method(
			$property_id,
			function ( Request $request ) use ( $raw_custom_dimension, $property_id ) {
				$url = parse_url( $request->getUri() );

				if ( "/v1beta/properties/$property_id/customDimensions" === $url['path'] ) {
					$custom_dimension = new GoogleAnalyticsAdminV1betaCustomDimension( $raw_custom_dimension );

					return new Response(
						200,
						array(),
						json_encode( $custom_dimension )
					);
				}

				return new Response( 200 );
			}
		);

		$this->grant_scope( Analytics_4::EDIT_SCOPE );

		$response = $this->analytics->set_data(
			'create-custom-dimension',
			array(
				'propertyID'      => $property_id,
				'customDimension' => $raw_custom_dimension,
			)
		);

		$this->assertNotWPError( $response );

		$this->assertEquals( $raw_custom_dimension, (array) $response->toSimpleObject() );
```

See the `test_sync_audiences()` function in the same file for reference.